snakemake_storage_plugin_s3/__init__.py

```
@@ -282,7 +285,15 @@ def exists(self) -> bool:
         try:
             self.s3obj().load()
         except botocore.exceptions.ClientError as e:
-            if e.response["Error"]["Code"] == "404":
+            err_code = e.response["Error"]["Code"]
+            if err_code == "400":
+                raise WorkflowError(
+                    f"Bad request for S3 object {self.query}. This can happen if "
+                    "the query contains unsupported characters (e.g. '..'), "
+                    "if your credentials are invalid, or if your permissions are "
+                    "insufficient."
+                )
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)